File: apina-java/src/test/java/fi/evident/apina/java/model/JavaMethodTest.java
package fi.evident.apina.java.model;

import fi.evident.apina.java.model.type.JavaBasicType;
import org.junit.Test;
import org.objectweb.asm.Opcodes;

import static java.util.Collections.emptyList;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class JavaMethodTest {

    @Test
    public void modifiers() {
        assertThat(methodWithModifiers(0).isStatic(), is(false));
        assertThat(methodWithModifiers(Opcodes.ACC_STATIC).isStatic(), is(true));
    }

    private static JavaMethod methodWithModifiers(int modifiers) {
        return new JavaMethod("foo", JavaVisibility.PUBLIC, new JavaBasicType("java.lang.String"), emptyList(), modifiers);
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/java/model/JavaMethodTest.kt
package fi.evident.apina.java.model

import fi.evident.apina.java.model.type.JavaType
import fi.evident.apina.java.model.type.TypeSchema
import org.junit.jupiter.api.Test
import org.objectweb.asm.Opcodes
import java.util.Collections.emptyList
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class JavaMethodTest {

    @Test
    fun modifiers() {
        assertFalse(methodWithModifiers(0).isStatic)
        assertTrue(methodWithModifiers(Opcodes.ACC_STATIC).isStatic)
    }

    private fun methodWithModifiers(modifiers: Int) =
        JavaMethod(
            owningClass = arbitraryClass(),
            name = "foo",
            visibility = JavaVisibility.PUBLIC,
            returnType = JavaType.basic<String>(),
            parameters = emptyList(),
            modifiers = modifiers,
            schema = TypeSchema()
        )

    private fun arbitraryClass() =
        JavaClass(
            type = JavaType.Basic("test.Bar"),
            superClass = JavaType.basic<Any>(),
            interfaces = emptyList(),
            modifiers = 0,
            schema = TypeSchema()
        )
}
